src/julia_gc.c

```diff
@@ -518,10 +521,64 @@ static void MarkFromList(PtrArray * arr)
 
 static TaskInfoTree * task_stacks = NULL;
 
+// We need to access the safe_restore member of the Julia TLS. Unfortunately,
+// its offset changes with the Julia version. In order to be able to produce
+// a single gap executable resp. libgap shared library which works across
+// multiple versions, we do the following:
+// - Julia 1.3 and 1.4 are the reference, the relative offset there hence is
+//   defined to be 0 (the absolute offset of safe_restore is 6840 on Linux and
+//   6816 on macOS)
+// - Julia 1.5 uses relative offset 8 (absolute offset is 6848 reps. 6824)
+// - Julia 1.6 added APIs to get and set the safe_restore value
+static int safe_restore_offset;
+
+static void set_safe_restore_with_offset(jl_jmp_buf * buf)
+{
+    jl_ptls_t tls = (jl_ptls_t)((char *)JuliaTLS + safe_restore_offset);
+    tls->safe_restore = buf;
+}
+
+static jl_jmp_buf * get_safe_restore_with_offset(void)
+{
+    jl_ptls_t tls = (jl_ptls_t)((char *)JuliaTLS + safe_restore_offset);
+    return tls->safe_restore;
+}
+
+static void (*set_safe_restore)(jl_jmp_buf * buf);
+static jl_jmp_buf * (*get_safe_restore)(void);
+
+static void SetupSafeRestoreHandlers(void)
+{
+    get_safe_restore = dlsym(RTLD_DEFAULT, "jl_get_safe_restore");
+    set_safe_restore = dlsym(RTLD_DEFAULT, "jl_set_safe_restore");
+
+    // if the new API is available, just use it!
+    if (get_safe_restore && set_safe_restore)
+        return;
+
+    GAP_ASSERT(!set_safe_restore && !Get_safe_restore);
```

Comment:
I think that `get_safe_restore` got capitalized by accident here. It only results in an error with `--enable-debug`, though.

Reply:
Oops. That also made me realize we don't have `--enable-debug` on the Julia tests, which would have caught that. I've added a commit to fix that, too.